src/libraries/Microsoft.Extensions.Primitives/src/StringSegment.cs

```diff
@@ -146,8 +146,7 @@ public bool HasValue
         /// </returns>
         public static int Compare(StringSegment a, StringSegment b, StringComparison comparisonType)
         {
-            int minLength = Math.Min(a.Length, b.Length);
-            int diff = string.Compare(a.Buffer, a.Offset, b.Buffer, b.Offset, minLength, comparisonType);
+            int diff = a.AsSpan().CompareTo(b.AsSpan(), comparisonType);
             if (diff == 0)
```

Comment:
This `if (diff == 0)` clause should also be removed.

Reply:
oops, I forgot to remove that, should be removed now.